### ISSUE TITLE
Resulting VoxelMorph Atlas is Blurry and Lacks Detail

### ISSUE THREAD

**Issue by siavashk** [voxelmorph/voxelmorph#587]
**Task** (what are you trying to do/register?)
I am trying to build a template for whole body sagittal T2 MRI from a dataset of ~1000 images. These are 2D images. Please see example below:
![image](https://github.com/voxelmorph/voxelmorph/assets/1020752/5ac5ac7c-c430-4e62-92f4-651ec975bea7)

**The issue is that the resulting Atlas is very blurry and lacks detail.**

**What have you tried**
I ran the code unconditional template for brain 2D data from [this colab](https://colab.research.google.com/drive/1SkQbrWTQHpQFrG4J2WoBgGZC9yAzUas2?usp=sharing#scrollTo=S-t3ztxZgpW1) on my dataset.

**Details of experiments**
Data: ~1000 male T2 whole body sagittal MRI. The background is set to zero and the foreground is normalized to `[0., 1.]` range.
Relevant code snippet:
```python
enc_nf = [16, 32, 32, 32]
dec_nf = [32, 32, 32, 32, 32, 16, 16]
model = vxm.networks.TemplateCreation(vol_shape, nb_unet_features=[enc_nf, dec_nf])

image_loss_func = vxm.losses.MSE().loss
neg_loss_func = lambda _, y_pred: image_loss_func(model.references.atlas_tensor, y_pred)
losses = [image_loss_func, neg_loss_func, vxm.losses.MSE().loss, vxm.losses.Grad('l2', loss_mult=2).loss]
loss_weights = [0.5, 0.5, 1, 0.01]

model.compile('adam', loss=losses, loss_weights=loss_weights)

gen = template_gen(...) # similar to template_gen in the notebook
hist = model.fit(gen, epochs=100, steps_per_epoch=25, verbose=0, callbacks=[tqdm_cb])
```

Results:
![image](https://github.com/voxelmorph/voxelmorph/assets/1020752/9d1e5dea-cb3b-44ee-a820-8842b48f05b0)

Atlas:
```python
%matplotlib widget
atlas = model.references.atlas_layer.get_weights()[0][..., 0]
plt.imshow(atlas, cmap='gray')
plt.axis('off');
```

As you see the atlas is very blurry.
![image](https://github.com/voxelmorph/voxelmorph/assets/1020752/890c5be5-2672-4ec6-8c0e-578a81b58a48)

**Comment by wilcamsdington** [comment]
I have also encountered the same issue

**Comment by siavashk** [comment]
@adalca I know that you are probably busy with MICCAI submission(s) but would you kindly have a look at this?

**Comment by adalca** [comment]
that's certainly way too blurry! what's the intensity range of your images? Cna you show me the atlas image of your initialization?

**Comment by siavashk** [comment]
The intensities are between [0., 1.]. Unfortunately I terminated the instance that had the code / data so it is difficult to reproduce the experiment. But I followed the exact same code as in [the example colab](https://colab.research.google.com/drive/1SkQbrWTQHpQFrG4J2WoBgGZC9yAzUas2?usp=sharing#scrollTo=S-t3ztxZgpW1) in Section **Unconditional Template (2D Brain slices)**.

If you don't know what is causing this off the top of your head, we can close the issue. I am just curious about the answer at this point.

**Comment by adalca** [comment]
Please see my answers in #589  -- and in particular [this tutorial](https://colab.research.google.com/drive/1SkQbrWTQHpQFrG4J2WoBgGZC9yAzUas2?usp=sharing#scrollTo=9OylH0vLjBNz).

**Comment by siavashk** [comment]
Thank you. Interesting conversation.
